File: prod/backend/WebApp/Data/Entities/Audit/ActionLog.cs
using WebApp.Data.Entities.Users;

namespace WebApp.Data.Entities.Audit;

public class ActionLog
{
    public Guid Id { get; set; }
    public Guid UserId { get; set; }
    public User User { get; set; }
    public DateTime DateTime { get; set; }
    public string? IP { get; set; }
    public string API { get; set; } = null!;
    public string? Note { get; set; }
}

File: prod/frontend/src/api/action-log.ts
import { api } from "@/lib/api";

export interface ActionLog {
  id: string;
  userId: string;
  dateTime: string;
  ip: string | null;
  api: string;
  note: string | null;
}

export interface PaginatedActionLogResponse {
  items: ActionLog[];
  totalCount: number;
  totalPages: number;
  currentPage: number;
  pageSize: number;
}

export interface ActionLogParams {
  page?: number;
  pageSize?: number;
}

export const actionLogApi = {
  getAll: async (
    params: ActionLogParams = {}
  ): Promise<PaginatedActionLogResponse> => {
    const { page = 1, pageSize = 25 } = params;
    const response = await api.get<PaginatedActionLogResponse>("/api/ActionLog", {
      params: { page, pageSize },
    });
    return response.data;
  },

  getById: async (id: string): Promise<ActionLog> => {
    const response = await api.get<ActionLog>(`/api/ActionLog/${id}`);
    return response.data;
  },
};
